qcodes/tests/drivers/test_keithley_s46.py

```diff
@@ -66,8 +70,9 @@ def test_query_close_once_at_init(caplog):
     Test that, during initialisation, we query the closed channels only once
     """
     with caplog.at_level(logging.DEBUG):
-        S46('s46_test_query_once', address='GPIB::2::INSTR', visalib=visalib)
+        inst = S46('s46_test_query_once', address='GPIB::2::INSTR', visalib=visalib)
         assert caplog.text.count(":CLOS?") == 1
+        inst.close()
```

Comment:
these calls to `close` (throughout the PR) should ideally should be taken care of by `request.addfinalizer(..)`. If we have more unforeseen issues in the future, i wouldn't want the absence of `request.addfinalizer(..)` be part of that issue.

Reply:
This test does not use a fixture. The tests that does used try/yield finally which will be equivalently good. The fix of this test and others that do not use fixtures is out of scope for this pr. To be solved probably this will need to resolve the general issue with exceptions raised in instrument construction leaving instruments in an incomplete state

Reply:
ok, ok) for later)